### PR TITLE
[SYCL] Add check for symbols table of DPC++ RT on Windows

### PR DESCRIPTION
**Pull request by dm-vodopyanov**
This patch adds the check for symbols table of DPC++ RT on Windows to
prevent ABI breaks on Windows

### CODE REVIEW

**Comment by dm-vodopyanov** [comment]
I'm working on resolving the CI failures.

File: sycl/test/abi/sycl_symbols_windows.dump [hunk 1]
```diff
@@ -0,0 +1,4550 @@
+################################################################################
+# This file is automatically generated by abi_check.py tool.
+# DO NOT EDIT IT MANUALLY. Refer to sycl/docs/ABIPolicyGuide.md for more info.
+################################################################################
+
+# RUN: env LLVM_BIN_PATH=%llvm_build_bin_dir python %sycl_tools_src_dir/abi_check.py --mode check_symbols --reference %s %sycl_libs_dir/sycl.dll
```
Comment (alexbatashev) [inline sycl/test/abi/sycl_symbols_windows.dump:6]:
```suggestion
# RUN: env LLVM_BIN_PATH=%llvm_build_bin_dir python %sycl_tools_src_dir/abi_check.py --mode check_symbols --reference %s %llvm_build_bin_dir/sycl.dll
```

Comment (dm-vodopyanov) [reply sycl/test/abi/sycl_symbols_windows.dump]: Fixed

**Review by alexbatashev — DISMISSED** [review]

File: sycl/test/tools/abi_check_positive_dump.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 __libc_csu_fini
 __libc_csu_init
+_dl_relocate_static_pie
```
Comment (dm-vodopyanov) [inline sycl/test/tools/abi_check_positive_dump.txt:3]:
```
Symbol {                                                                                                                                                                                                               
    Name: _dl_relocate_static_pie (315)                                                                                                                                                                                  
    Value: 0x4003E0                                                                                                                                                                                                      
    Size: 2                                                                                                                                                                                                              
    Binding: Global (0x1)                                                                                                                                                                                                
    Type: Function (0x2)                                                                                                                                                                                                 
    Other [ (0x2)                                                                                                                                                                                                        
      STV_HIDDEN (0x2)                                                                                                                                                                                                   
    ]                                                                                                                                                                                                                    
    Section: .text (0xA)                                                                                                                                                                                                 
  }            
```

**Review by alexbatashev — APPROVED** [review]

**Pull request merged**